controllers/awsadapterconfig_controller.go

```diff
@@ -444,6 +447,58 @@ func (r *AWSAdapterConfigReconciler) updateLastPollStatusFailure(ctx context.Con
 	return ctrl.Result{RequeueAfter: r.RequeueInterval}, nil
 }
 
+func (r *AWSAdapterConfigReconciler) IsAWSAdapterConfigPresent() (bool, error) {
+	obj := &securityv1alpha1.AWSAdapterConfig{}
+	err := r.Get(context.TODO(), apimachineryTypes.NamespacedName{Namespace: getAdapterNamespace(), Name: getAdapterName()}, obj)
+	if err == nil {
+		return true, nil
+	}
+	if errors.IsNotFound(err) {
+		return false, nil
+	}
+	return false, err
+}
+
+func (r *AWSAdapterConfigReconciler) CreateAWSAdapterConfig() error {
+	clusterName := getClusterName()
+	clusterRegion := getClusterRegion()
+	adapterName := getAdapterName()
+	adapterNamespace := getAdapterNamespace()
+	return r.Create(context.TODO(), &securityv1alpha1.AWSAdapterConfig{
```

Comment:
If any of the above are empty, then don't create.

Reply:
Done. I moved the required parameters and their error handling to the main file, so that we can right away return if all required params are not present.